syncer/syncer.go

```diff
@@ -405,19 +410,13 @@ func (s *Syncer) Init() (err error) {
 
 // initShardingGroups initializes sharding groups according to source MySQL, filter rules and router rules
 // NOTE: now we don't support modify router rules after task has started
-func (s *Syncer) initShardingGroups(conn *DBConn) error {
+func (s *Syncer) initShardingGroups() error {
 	// fetch tables from source and filter them
 	sourceTables, err := s.fromDB.fetchAllDoTables(s.bwList)
 	if err != nil {
 		return err
 	}
 
-	// clear old sharding group and initials some needed data
-	err = s.sgk.Init(conn)
```

Comment:
why do we extract `s.sgk.Init` from `initShardingGroups`? simply for testing?

Reply:
yes, for testing, may be we can add a new function such like `initSGK` which include `s.sgk.Init` and ` initShardingGroups`